### ISSUE TITLE
Can't use module_override to override a repo fetched inside module extension

### ISSUE THREAD

**Issue by peakschris** [bazelbuild/bazel#23580]
### Description of the bug:

We have a module extension that uses a repo rule with rctx.download to download an artifact. We use it from the top level MODULE.bazel as:

```
fetcher = use_extension("//src/repo/bazel/rules:fetcher_extension.bzl", "fetcher")
fetcher.artifact_with_plat(
     name = "artifact1",
     url = y
     hash = z
)
use_repo("artifact1")
```

This works fine. The artifact ends up in my extensions folder as `+fetcher+artifact1`

How do I then override this artifact using override_module? `--override_module=artifact1=...` does not work, and `--override_module=+fetcher+artifact1=...` does not work.

### Which category does this issue belong to?

External Dependency

### Which operating system are you running Bazel on?

windows

### What is the output of `bazel info release`?

release 7.3.1

### Have you found anything relevant by searching the web?

no

**Comment by fmeum** [comment]
This will be solved by https://github.com/bazelbuild/bazel/pull/23534.

**Comment by peakschris** [comment]
@fmeum excellent! Can #23534 be released in bazel 7.3.x or 7.4?

**Comment by fmeum** [comment]
It will be in 7.4.0.

**Comment by iancha1992** [comment]
A fix for this issue has been included in [Bazel 7.4.0 RC1](https://github.com/bazelbuild/bazel/releases/tag/7.4.0rc1). Please test out the release candidate and report any issues as soon as possible.
If you're using Bazelisk, you can point to the latest RC by setting `USE_BAZEL_VERSION=7.4.0rc1`. Thanks!
